app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -11,19 +11,130 @@ import 'package:github/github.dart';
 import '../proto/internal/scheduler.pb.dart' as pb;
 import 'target.dart';
 
+/// Key used for selecting which .ci.yaml to use
+enum CiType {
+  /// "Default" associated with this slug / ci.yaml
+  ///
+  /// Pre-fusion: engine, framework, etc
+  /// Post-fusion: framework, etc
+  ///
+  /// NOTE: Required to be present
+  any,
+
+  /// Engine's .ci.yaml file located in the monorepo
+  fusionEngine,
+}
+
+/// Wrapper arround one or more [CiYaml]
+class CiYaml {
+  CiYaml({
+    required this.slug,
+    required this.branch,
+    required Map<CiType, pb.SchedulerConfig> yamls,
+    CiYaml? totConfig,
+    bool validate = false,
+    this.isFusion = false,
+  }) {
+    for (final MapEntry(key: type, value: config) in yamls.entries) {
+      configs[type] = CiYamlInner(
+        slug: slug,
+        branch: branch,
+        config: config,
+        type: type,
+        validate: validate,
+        isFusion: isFusion,
+        totConfig: totConfig?.configs[type],
+      );
+    }
+  }
+
+  final bool isFusion;
+
+  final configs = <CiType, CiYamlInner>{};
+
+  /// Get's the [pb.SchedulerConfig] for the requested [type].
+  ///
+  /// The type is expected to exist and will fail otherwise.
+  pb.SchedulerConfig configForInner(CiType type) => configs[type]!.config;
+
+  /// Get's the [CiYamlInner] for the requested [type].
+  ///
+  /// The type is expected to exist and will fail otherwise.
+  CiYamlInner innerFor(CiType type) => configs[type]!;
+
+  /// The [RepositorySlug] that [config] is from.
+  final RepositorySlug slug;
+
+  /// The git branch currently being scheduled against.
+  final String branch;
```

Comment:
Feels wrong for something that represents the ci.yaml config to be "scheduled against" some branch. Can we separate the scheduling concern from the configuration?

Reply:
This is copy-pasta from the old CiYaml and I don't want to make big logical changes at the moment. My understanding is this is specifically for release branch handling - and it looks inside the ci yaml for special bits.